src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInMetricsListener.cs

```diff
@@ -130,6 +171,7 @@ public void OnEventWritten_Old(string name, object payload)
 
     public void OnEventWritten_New(string name, object payload)
     {
+        var activity = Activity.Current;
```

Comment:
Let's add a `Debug.Assert(activity != null)` here. I'm paranoid 🤣

Reply:
we have a check [here](https://github.com/open-telemetry/opentelemetry-dotnet/blob/37481f1ee608a7781c2b4693b9bb5891bc767c11/src/Shared/DiagnosticSourceInstrumentation/DiagnosticSourceListener.cs#L46-L49)